flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointProperties.java

```diff
@@ -248,6 +248,15 @@ public String toString() {
 			false,
 			false);
 
+	private static final CheckpointProperties SYNC_SAVEPOINT_NOT_FORCED = new CheckpointProperties(
+			false,
```

Comment:
Have you thought about implications of setting this flag to `false` in other places? like `PendingCheckpoint#canBeSubsumed()`? or `PendingCheckpoint#assertAbortSubsumedForced`?

Reply:
You're right, I missed this use of  the `force` flag.
Fixed by replacing `isForce` by `isSavepoint` in these methods.

Reply:
Have you added a test coverage for those bugs?

Reply:
This is covered by `PendingCheckpointTest.estCanBeSubsumed`.